Helper/SiteAccess.php

```diff
@@ -0,0 +1,248 @@
+    }
+
+    /**
+     * Checks if siteAccessName is the same as system default siteAccess name.
+     *
+     * @param string $siteAccessName
+     *
+     * @return bool
+     */
+    private function isSiteAccessSameAsSystemDefault($siteAccessName)
+    {
+        return $siteAccessName === self::SYSTEM_DEFAULT_SITEACCESS_NAME;
+    }
+
+    /**
+     * Checks if mandatorId is configured with default siteAccess.
+     *
+     * @param int $mandatorId
+     *
+     * @return bool
+     */
+    private function isMandatorIdConfigured($mandatorId)
+    {
+        return in_array($this->defaultSiteAccessName, $this->siteAccessConfig)
+            && $this->siteAccessConfig[$this->defaultSiteAccessName]['yoochoose']['customer_id'] == $mandatorId;
+    }
+
+    /**
+     * Returns siteAccesses based on mandatorId, requested siteAccess or default SiteAccess.
+     *
+     * @param null|int $mandatorId
+     * @param null|string $siteAccess
+     *
+     * @return array
+     */
+    public function getSiteAccesses($mandatorId = null, $siteAccess = null)
+    {
+        if ($mandatorId) {
+            $siteAccesses = $this->getSiteaccessesByMandatorId($mandatorId);
+        } elseif ($siteAccess) {
+            $siteAccesses = array($siteAccess);
+        } else {
+            $siteAccesses = array($this->siteAccess->name);
+        }
+
+        return $siteAccesses;
+    }
+
+    /**
+     * Returns Recommendation Service credentials based on current siteAccess or mandatorId.
+     *
+     * @param null|int $mandatorId
+     * @param null|string $siteAccess
+     *
+     * @return array
+     */
+    public function getRecommendationServiceCredentials($mandatorId = null, $siteAccess = null)
+    {
+        $siteAccesses = $this->getSiteAccesses($mandatorId, $siteAccess);
+        $siteAccess = end($siteAccesses);
+
+        if ($siteAccess === self::SYSTEM_DEFAULT_SITEACCESS_NAME) {
+            $siteAccess = null;
+        }
+
+        $customerId = $this->configResolver->getParameter('yoochoose.customer_id', 'ez_recommendation', $siteAccess);
+        $licenceKey = $this->configResolver->getParameter('yoochoose.license_key', 'ez_recommendation', $siteAccess);
+
+        return [$customerId, $licenceKey];
+    }
+
+    /**
+     * Returns main languages from siteAccess list.
+     *
+     * @param array $siteAccesses
+     *
+     * @return array
+     */
+    private function getMainLanguagesBySiteAccesses($siteAccesses)
+    {
+        $languages = array();
+
+        foreach ($siteAccesses as $siteAccess) {
+            $languageList = $this->configResolver->getParameter(
+                'languages',
+                '',
+                $siteAccess !== 'default' ? $siteAccess : null
+            );
+            $mainLanguage = reset($languageList);
+
+            if ($mainLanguage) {
+                $languages[$mainLanguage] = $mainLanguage;
+            }
+        }
+
+        return array_keys($languages);
```

Comment:
how about just `return !empty($languageList) ? [$mainLanguage] : []` a couple of lines above?

Reply:
I think the logic will be different. The purpose for this method is to collect all of the first languages from passed siteAccesses list, not only the first one, so `return` should not be inside `foreach`